src/Supervisor.ts

```diff
@@ -136,21 +139,35 @@ class Supervisor {
         ssh = connectionPool[job.id].ssh;
       }
 
+      var exit = false;
+      var wait = 2;
+
       // connect ssh & run
-      try {
-        if (!ssh.connection.isConnected())
-          await ssh.connection.connect(ssh.config);
-        await ssh.connection.execCommand("echo"); // test connection
-        if (job.maintainerInstance.isInit) {
-          await job.maintainerInstance.maintain();
-        } else {
-          await job.maintainerInstance.init();
+      while (!exit && !ssh.connection.isConnected()) {
+        if (wait > 100) {
+          console.log("jdebug bad");
+          exit = true;
+          continue;
+        }
+        try {
+          console.log("jdebug ok");
+          await sleep(wait * 1000);
```

Comment:
`await sleep` will block the execution of the job as they are executed sequentially. A more appropriate method might be to wrap the `createMaintianerWorker` at line 113 `this.createMaintainerWorker(job);` within a Promise, use `setTimeout` as the timer, and interrupt the execution using `reject`:

```JavaScript
   new Promise((resolve, reject) => {
        console.log("Function starts");

        setTimeout(() => {
            console.log("setTimeout callback runs");
            reject('Interrupted by setTimeout');
        }, 1000);

        // Simulate some work
        for (let i = 0; i < 5; i++) {
            console.log(`Processing ${i}`);
        }

        resolve('Completed successfully');
    });
```

Reply:
Also, you can use this to emit a failure event:

```JavaScript
          self.emitter.registerEvents(
            job,
            "JOB_REGISTERED",
            `job [${job.id}] is registered with the supervisor, waiting for initialization`
          );
```